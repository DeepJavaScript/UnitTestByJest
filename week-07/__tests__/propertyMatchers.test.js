describe('Property Matchers', () => {
  // it('will fail every time', () => {
  //   const user = {
  //     name: 'Titan',
  //     createdAt: new Date(),
  //     id: Math.floor(Math.random() * 20)
  //   };

  //   console.log(user);
  
  //   expect(user).toMatchSnapshot();
  // });

  it('will check the matchers and pass', () => {
    const user = {
      name: 'Titan',
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20)
    };

    console.log(user);
  
    expect(user).toMatchSnapshot({
      name: 'Titan',
      createdAt: expect.any(Date),
      id: expect.any(Number),
    });
  });

  it('will check the values and pass', () => {
    const user = {
      name: 'Titan',
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20)
    };

    console.log(user);
  
    expect(user).toMatchSnapshot({
      name: 'Titan',
      createdAt: expect.any(Date),
      id: expect.any(Number),
    });
  });
});