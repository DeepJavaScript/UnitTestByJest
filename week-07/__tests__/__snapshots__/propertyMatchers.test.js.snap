// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Matchers will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "Titan",
}
`;

exports[`Property Matchers will check the values and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "Titan",
}
`;
